openwrt-openwisp-monitoring/files/netjson-monitoring.lua

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env lua
+-- retrieve monitoring information
+-- and return it as NetJSON Output
+local cjson = require('cjson')
+
+local ubus_lib = require('ubus')
+local ubus = ubus_lib.connect()
+if not ubus then
+    error('Failed to connect to ubusd')
+end
+
+local dhcp = require('dhcp')
+local interface_functions = require('interfaces')
+local neighbors_functions = require('neighbors')
+local resources = require('resources')
+local utils = require('utils')
+local wifi = require('wifi')
+
+-- helpers
+local iwinfo_modes = {
+    ['Master'] = 'access_point',
+    ['Client'] = 'station',
+    ['Mesh Point'] = '802.11s',
+    ['Ad-Hoc'] = 'adhoc'
+}
+
+-- collect system info
+local system_info = ubus:call('system', 'info', {})
+local board = ubus:call('system', 'board', {})
+local loadavg_output = io.popen('cat /proc/loadavg'):read()
+loadavg_output = utils.split(loadavg_output, ' ')
+local load_average = {tonumber(loadavg_output[1]), tonumber(loadavg_output[2]), tonumber(loadavg_output[3])}
+
+
+-- init netjson data structure
+local netjson = {
+    type = 'DeviceMonitoring',
+    general = {
+        hostname = board.hostname,
+        local_time = system_info.localtime,
+        uptime = system_info.uptime
+    },
+    resources = {
+        load = load_average,
+        memory = system_info.memory,
+        swap = system_info.swap,
+        cpus = resources.get_cpus(),
+        disk = resources.parse_disk_usage()
+    }
+}
+
+local dhcp_leases = dhcp.get_dhcp_leases()
+if not utils.is_table_empty(dhcp_leases) then
+    netjson.dhcp_leases = dhcp_leases
+end
+
+local neighbors = neighbors_functions.get_neighbors()
```

Comment:
Used `_functions` here because there is already a variable with same name

Reply:
Suggested variable identifier `local host_neighbors = neighbors_functions.get_neighbors()`